src/util/util_creds.h

```diff
@@ -71,5 +77,19 @@ struct cli_creds {
 
 #define cli_creds_get_uid(x) (x->ucred.uid)
 #define cli_creds_get_gid(x) (x->ucred.gid)
+#endif
```

Comment:
(1) We need to have `cli_creds` defined for both UCRED and XUCRED cases - it is used in 'responder/kcm'.

(2) We don't need both `CRED_*ID` and `cli_creds_get_*id` and I think we don't use `CRED_PID` currently.

(3) I suggest to structure code like:
```
#ifdef HAVE_UCRED
...
#elif HAVE_XUCRED
...
#endif
```

Reply:
2) How is so? `cli_creds_get_*` are getters for `cli_creds`, while `CRED_*ID` accept plain `STRUCT_CRED`.

Reply:
Something like
```
struct cli_creds {
    STRUCT_CRED ucred;
...
};

#define cli_creds_get_uid(x) (CRED_UID(x->ucred))
```